Add mlt status command

dmsuehir opened the pull request:
fixes #127

Adds `mlt status` command.  

Finding the pods associated with a job can be different based on which template is used.  For example, with hello-world we can look for pods with a label `job-name=...`.  But for `TFJob` templates, we instead need to look for pods with a label `tf_job_name=...`.  Also, since the TFJob pods are terminated after the job completes, we should also show list the `TFJob` in the status (otherwise the status doesn't show anything after the job completes).  Since these commands are different per template, I added a `status` rule to the `Makefile` for each template.  And in case the user has a template that doesn't have the `status` rule in the `Makefile` (like maybe they have an old app from before this was added), then I have a default way to grab status based on pod name.

## Reviewer Checklist

 - [ ] Do you understand it?
 - [ ] Is it correct?
 - [ ] Is it safe?
 - [ ] Is it legally compliant?
 - [ ] Is it robust?
 - [ ] Is it simple?
 - [ ] Is it tested?
 - [ ] Is it documented?
 - [ ] Is the style consistent?

See [the reviewer guide](docs/reviews.md) for more detail on each check.

nqn commented on mlt/commands/status.py:
```diff
@@ -0,0 +1,94 @@
+
+        # Get pods list and display it
+        cmd = "kubectl get pods -a -o wide --namespace {} | " \
+              "awk 'NR==1 || /{}/'".format(namespace, pod_name_prefix)
```
Let's try to the parsing in python and not pipe to `awk`, _and_ get rid of `shell=True`. In general, assume that any subprocess call is as sensitive as sql query strings. Someone could set the namespace in the config file to `&& rm -rf /` or something worse.

nqn commented on mlt/commands/status.py:
```diff
@@ -0,0 +1,94 @@
+                print("Error while getting app status: {}".format(e.output))
+
+    @staticmethod
+    def _default_status(namespace, app_name, app_run_id):
```
How about we just don't do anything if there isn't a target for it?

dmsuehir replied:
@nqn  My thought was that if someone had initialized an app prior to `mlt status` existing, then their app won't have the `status` target in the Makefile, so I figured some default behavior to get the pods for their job would at least get them some status information.  But, if we are ok with not supporting that use case, I am fine with removing this and instead printing a message about the command being unsupported for their app.  What we really need is #75 so that they can grab and update.  :)

dmsuehir commented:
@nqn I made some updates:
* Removed the "default" status behavior (when the target isn't found in the Makefile) and instead just print a message saying that the app does doesn't support the status command.
* Removed the check for the specific makefile error and instead handle it from the Makefile itself.
* Remove the `shell=True`
